### PR TITLE
Feat: Notifications Banner

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
Jira: https://mattermost.atlassian.net/browse/MM-60477

### CODE REVIEW

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+5. Click on "Enable desktop notification".
+6. Verify a confirmation pop up is shown with message `Mattermost(<Instance link>)  would like to send you notifications.`.
+7. **Allow** the notifications permissions.
+8. Receive a notification ping from another user.
+9. Verify the notification pops up and also can be seen in the Notification tray or Notification centre on WINDOWS and MAC.
+10. Verify the Global banner is not visible anymore.
+11. Open Settings modal. Expand "Desktop and mobile notification section".
+12. Verify "Desktop notifications aren’t enabled" message is not visible anymore.
+
+### Scenario: User Deny's the desktop notifications on WINDOWS or MAC OS
```
[the remark this answers is not in the document]
Comment (yasserfaraazkhan) [reply data/test-cases/notifications-banner/1-Enable-disable-notif.md]: done

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md
[the remark this answers is not in the document]
Comment (yasserfaraazkhan) [reply data/test-cases/notifications-banner/1-Enable-disable-notif.md:1]: Got it. Added Chrome, Firefox, Edge and Safari

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+5. Click on "Enable desktop notification".
+6. Verify a confirmation pop up is shown with message `Mattermost(<Instance link>)  would like to send you notifications.`.
+7. **Allow** the notifications permissions.
+8. Receive a notification ping from another user.
+9. Verify the notification pops up and also can be seen in the Notification tray or Notification centre on WINDOWS and MAC.
+10. Verify the Global banner is not visible anymore.
+11. Open Settings modal. Expand "Desktop and mobile notification section".
+12. Verify "Desktop notifications aren’t enabled" message is not visible anymore.
+
+### Scenario: User Deny's the desktop notifications on WINDOWS / MAC / Ubuntu OS
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/1-Enable-disable-notif.md:55]:
```suggestion
### Scenario: User Denies the desktop notifications on WINDOWS / MAC / Ubuntu OS
```

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+1. Log into User's account.
+2. Verify desktop app shows the global blue banner with message "We need your permission to show desktop notifications." and the button "Enable desktop notification".
+3. Open Settings modal. Expand "Desktop and mobile notification section".
+4. In Notifications section, there should be a message "Desktop notifications aren’t enabled" and the button "Enable desktop notification".
+5. Click on "Enable desktop notification".
+6. Verify a confirmation pop up is shown with message `Mattermost(<Instance link>)  would like to send you notifications.`.
+7. **Deny** the notifications permissions.
+8. Receive a notification ping from another user.
+9. Verify the notification pop-up are not seen.
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/1-Enable-disable-notif.md:67]:
```suggestion
9. Verify the notification pop-up is not seen.
```

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+10. Verify the Global banner is not visible anymore.
+11. Open Settings modal.
+12. Verify a Alert tag is visible, saying "Permission required".
+13. Expand "Desktop and mobile notification section".
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/1-Enable-disable-notif.md:71]:
```suggestion
13. Expand "Desktop and mobile notifications" section.
```

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+14. Verify "Desktop notifications permission was denied" message is visible. A buttons with text "Enable desktop notifications" and "How to enable notifications" are visible.
+
+### Scenario: User Deny's the desktop notifications on Browsers Chrome/Safari/Firefox/Edge
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/1-Enable-disable-notif.md:74]:
```suggestion
### Scenario: User Denies the desktop notifications on browsers Chrome/Safari/Firefox/Edge
```

File: data/test-cases/notifications-banner/2-Mobile-disable-notif.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+# (Required) Ensure all values are filled up
+name: "Enable/Disabled notification on Mobile app"
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/2-Mobile-disable-notif.md:3]:
```suggestion
name: "Enable/Disable notification on Mobile app"
```

File: data/test-cases/notifications-banner/2-Mobile-disable-notif.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+status: Active
+priority: Normal
+folder: Notifications Banner
+authors: "@yasserfaraazkhan"
+team_ownership:
+- ICU
+priority_p1_to_p4: P2 - Core Functions (Do core functions work?)
```
Comment (lindalumitchell) [inline data/test-cases/notifications-banner/2-Mobile-disable-notif.md:10]:
```suggestion
priority_p1_to_p4: P3 - Deep Functions (Do extensive scenarios work?)
```

**Review by lindalumitchell — APPROVED** [review]
Nice, Yasser, thanks! Just a few nit-picky suggestions. :)

File: data/test-cases/notifications-banner/1-Enable-disable-notif.md (hunk 2 above)
Comment (yasserfaraazkhan) [reply data/test-cases/notifications-banner/1-Enable-disable-notif.md:55]: @lindalumitchell  Haha. I cant belive I wrote "Deny's"

I was hungry 🌯

**Pull request merged**